src/amms.py

```diff
@@ -7,297 +7,609 @@
+                continue
+
+            self.pools.append((ticker, pool))
+
+        LOG.info(f"Fetched BonkSwap pools: {[i[0] for i in self.pools]}")
+
+    def store_pool(self, pool: tuple[str, BonkPool]):
+
+        ticker, pool_info = pool
+
+        with get_db_session() as session:
+            entry = AmmLiquidity(
+                timestamp=self.timestamp,
+                dex=self.DEX_NAME,
+                pair=ticker,
+                market_address=self.TICKERS[ticker]["market_address"],
+                token_x=pool_info.token_x_reserve.v,
+                token_y=pool_info.token_y_reserve.v,
+                token_x_decimals=self.TICKERS[ticker]["token_x_decimals"],
+                token_y_decimals=self.TICKERS[ticker]["token_y_decimals"],
+                additional_info="{}",
+            )
+
+            # Add to session
+            session.add(entry)
+
+            #  Commit
+            session.commit()
+
+
+class DooarAMM(Amm):
+    DEX_NAME = "DOOAR"
+    TICKERS = {
+        "SOL-USDC": {
+            "token_x_account": Pubkey.from_string(
+                "GVfKYBNMdaER21wwuqa4CSQV8ajVpuPbNZVV3wcuKWhE"
+            ),
+            "token_y_account": Pubkey.from_string(
+                "ARryk4nSoS6bu7nyv6BgQah8oU23svFm7Rek7kR4fy3X"
+            ),
+        }
+    }
+
+    def get_pools(self):
+        self.pools: list[tuple[str, dict[str, Any]]] = []
+        client = Client(AUTHENTICATED_RPC_URL)
+
+        for ticker, pool_info in self.TICKERS.items():
+            token_x_acc = client.get_account_info_json_parsed(
+                pool_info["token_x_account"]
+            )
+            token_y_acc = client.get_account_info_json_parsed(
+                pool_info["token_y_account"]
+            )
+
+            if not token_x_acc.value:
+                LOG.warning(f"No value found for X of {self.DEX_NAME}:{ticker}")
+                continue
+
+            if not token_y_acc.value:
+                LOG.warning(f"No value found for Y of {self.DEX_NAME}:{ticker}")
+                continue
+
+            token_x_data = token_x_acc.value.data
+            token_y_data = token_y_acc.value.data
+
+            if not isinstance(token_x_data, ParsedAccount):
+                LOG.warning(f"Unable to parse X of {self.DEX_NAME}:{ticker}")
+                continue
+
+            if not isinstance(token_y_data, ParsedAccount):
+                LOG.warning(f"Unable to parse Y of {self.DEX_NAME}:{ticker}")
+                continue
+
+            token_x_parsed: dict = token_x_data.parsed
+            token_y_parsed: dict = token_y_data.parsed
+
+            token_x_amt = token_x_parsed["info"]["tokenAmount"]["amount"]
+            token_y_amt = token_y_parsed["info"]["tokenAmount"]["amount"]
+
+            token_x_decimals = token_x_parsed["info"]["tokenAmount"]["decimals"]
+            token_y_decimals = token_y_parsed["info"]["tokenAmount"]["decimals"]
+
+            self.pools.append(
```

Comment:
Can we have a unified type for an item of `self.pools`? It's a bit difficult to read the code when it changes across dexes.